packages/tools/src/tools/segmentation/strategies/BrushStrategy.ts

```diff
@@ -0,0 +1,289 @@
+    } else {
+      strategySpecificConfiguration.centerIJK = centerIJK;
+    }
+
+    this._fill.forEach((func) => func(enabledElement, initializedData));
+
+    const { segmentationVoxelValue, previewVoxelValue, previewSegmentIndex } =
+      initializedData;
+
+    triggerSegmentationDataModified(
+      initializedData.segmentationId,
+      segmentationVoxelValue.getArrayOfSlices()
+    );
+
+    // We are only previewing if there is a preview index, and there is at
+    // least one slice modified
+    if (!previewSegmentIndex || !previewVoxelValue.modifiedSlices.size) {
+      return null;
+    }
+    return initializedData;
+  };
+
+  protected createInitialized(
+    enabledElement: Types.IEnabledElement,
+    operationData: OperationData
+  ): InitializedOperationData {
+    const { viewport } = enabledElement;
+    const data = getStrategyData({ operationData, viewport });
+
+    if (!data) {
+      console.warn('No data found for BrushStrategy');
+      return operationData.preview;
+    }
+
+    if (isVolumeSegmentation(operationData)) {
+      const { referencedVolumeId, volumeId } =
+        operationData as LabelmapToolOperationDataVolume;
+
+      const imageVolume = cache.getVolume(referencedVolumeId);
+      const segmentation = cache.getVolume(volumeId);
+
+      if (
+        !csUtils.isEqual(segmentation.dimensions, imageVolume.dimensions) ||
+        !csUtils.isEqual(segmentation.direction, imageVolume.direction)
+      ) {
+        throw new Error(
+          'Only source data the same dimensions/size/orientation as the segmentation currently supported.'
+        );
+      }
+    }
+
+    const { imageVoxelValue, segmentationVoxelValue, segmentationImageData } =
+      data;
+    const previewVoxelValue =
+      operationData.preview?.previewVoxelValue ||
+      VoxelValue.historyVoxelValue(segmentationVoxelValue);
+
+    const previewSegmentIndex = operationData.previewColors ? 255 : undefined;
+    console.log('Using previewSegmentIndex', previewSegmentIndex);
+    const initializedData: InitializedOperationData = {
+      ...operationData,
+      previewSegmentIndex,
+      enabledElement,
+      imageVoxelValue,
+      segmentationVoxelValue,
+      segmentationImageData,
+      previewVoxelValue,
+      viewport,
+
+      centerWorld: null,
+      brushStrategy: this,
+    };
+
+    this._createInitialized.forEach((func) =>
+      func(enabledElement, initializedData)
+    );
+
+    return initializedData;
+  }
+
+  public initDown: (
+    enabledElement: Types.IEnabledElement,
+    operationData: OperationData
+  ) => void;
+
+  public completeUp: (
+    enabledElement: Types.IEnabledElement,
+    operationData: OperationData
+  ) => void;
+
+  public rejectPreview: (
+    enabledElement: Types.IEnabledElement,
+    operationData: OperationData
+  ) => void;
+
+  public acceptPreview: (
+    enabledElement: Types.IEnabledElement,
+    operationData: OperationData
+  ) => void;
+
+  public preview: (
```

Comment:
a method called preview (noun) does not ring a bell to me, accept/reject preview is quite clear, but `preview` seems unusuall, maybe rename it